src/qlever-petrimaps/GeomCache.cpp

```diff
@@ -318,6 +317,14 @@ void GeomCache::parse(const char* c, size_t size) {
   }
 }
 
+double GeomCache::getLoadStatusPercent() {
+  if (_totalSize == 0) {
+    return 0.0;
+  } else {
+    return (static_cast<double>(_curRow) / static_cast<double>(_totalSize) * 100);
```

Comment:
I think it would be better if _curRow was now a `std::atomic<size_t>`, as concurrent read/writes can now happen.